test/parallel/test-async-wrap-getasyncid.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const providers = Object.assign({}, process.binding('async_wrap').Providers);
+
+// Make sure that all Providers are tested.
+{
+  const hooks = require('async_hooks').createHook({
+    init(id, type) {
+      if (type === 'NONE')
+        throw new Error('received a provider type of NONE');
+      delete providers[type];
+    },
+  }).enable();
+  process.on('beforeExit', common.mustCall(() => {
+    process.removeAllListeners('uncaughtException');
+    hooks.disable();
+    delete providers.NONE;  // Should never be used.
+    const obj_keys = Object.keys(providers);
+    if (obj_keys.length > 0)
+      process._rawDebug(obj_keys);
+    assert.strictEqual(obj_keys.length, 0);
+  }));
+}
+
+function testUninitialized(req, ctor_name) {
+  assert.strictEqual(typeof req.getAsyncId, 'function');
+  assert.strictEqual(req.getAsyncId(), -1);
+  assert.strictEqual(req.constructor.name, ctor_name);
+}
+
+function testInitialized(req, ctor_name) {
+  assert.strictEqual(typeof req.getAsyncId, 'function');
+  assert(Number.isSafeInteger(req.getAsyncId()));
+  assert(req.getAsyncId() > 0);
+  assert.strictEqual(req.constructor.name, ctor_name);
+}
+
+
+{
+  const cares = process.binding('cares_wrap');
+  const dns = require('dns');
+
+  testUninitialized(new cares.GetAddrInfoReqWrap(), 'GetAddrInfoReqWrap');
+  testUninitialized(new cares.GetNameInfoReqWrap(), 'GetNameInfoReqWrap');
+  testUninitialized(new cares.QueryReqWrap(), 'QueryReqWrap');
+
+  testInitialized(dns.lookup('www.google.com', () => {}), 'GetAddrInfoReqWrap');
+  testInitialized(dns.lookupService('::1', 22, () => {}), 'GetNameInfoReqWrap');
+  testInitialized(dns.resolve6('::1', () => {}), 'QueryReqWrap');
+}
+
+
+{
+  const FSEvent = process.binding('fs_event_wrap').FSEvent;
+  testInitialized(new FSEvent(), 'FSEvent');
+}
+
+
+{
+  const JSStream = process.binding('js_stream').JSStream;
+  testInitialized(new JSStream(), 'JSStream');
+}
+
+
+if (common.hasCrypto) {
+  const tls = require('tls');
+  // SecurePair
+  testInitialized(tls.createSecurePair().ssl, 'Connection');
```

Comment:
AFAIK the `node::Connection` class can only be accessed via `tls.createSecurePair()`. If there's a way I'm missing I'd be happy to use it.